src/widgets/password-text.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Localized } from "fluent-react";
+import PropTypes from "prop-types";
+import React from "react";
+
+import { classNames } from "../common";
+
+import styles from "./input.css";
+
+const PASSWORD_DOT = "\u2022";
+
+export default class PasswordText extends React.Component {
+  static get propTypes() {
+    return {
+      className: PropTypes.string,
+      value: PropTypes.string,
+    };
+  }
+
+  static get defaultProps() {
+    return {
+      className: "",
+      value: "",
+    };
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showPassword: false,
+    };
+  }
+
+  showPassword(show) {
+    this.setState({showPassword: show});
+  }
+
+  render() {
+    const {className, value, ...props} = this.props;
+    const {showPassword} = this.state;
+
+    return (
+      <div {...props} className={classNames([
+        styles.monospace,
+        styles.passwordText,
```

Comment:
Two issues I ran into here, trying to handle [a "long password" case](https://app.zeplin.io/project/5c1829b8fde4d550a26b270f/screen/5c3d0105a8ecedbdc31cdb22) here:
1. How long is long? (e.g. 15 characters?) We could possibly switch a classname on value length
2. The whole form is on a shared CSS grid, with the copy buttons aligned. Breaking that alignment calls for a different overall layout for the form, I think? My CSS-fu fails me here.

Reply:
@mozilla-lockbox/ux some insights here would be helpful

>     1. How long is long? (e.g. 15 characters?) We could possibly switch a classname on value length

I would start with 15 or 20 personally, too

>     2. The whole form is on a shared CSS grid, with the copy buttons aligned. Breaking that alignment calls for a different overall layout for the form, I think? My CSS-fu fails me here.

I think the intent of the designs is for the reveal button to be adjacent to the password text, which means it's position could shift based on how long the value is.

To keep the reveal in a fixed spot, I think you're right that it would require changing the CSS grid column count.

Reply:
Well, I've got the reveal button sitting at the end of the text, which seems to match the spec. It's the unaligned copy button and expanding the width of the field that seems a problem

Reply:
sorry, I was misunderstanding completely!  And it looks like the UX questions are worked out.